app/auto_responder.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	l4g "github.com/alecthomas/log4go"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+func (a *App) SendAutoResponse(channel *model.Channel, receiver *model.User) {
+	if receiver == nil || receiver.NotifyProps == nil {
+		return
+	}
+
+	active, message :=
+		receiver.NotifyProps["auto_reply_active"] == "true",
+		receiver.NotifyProps["auto_reply_message"]
+
+	if active && message != "" {
+		autoResponsePost := &model.Post{
+			ChannelId: channel.Id,
+			Message:   message,
+			Type:      model.POST_AUTO_RESPONSE,
+			UserId:    receiver.Id,
+		}
+
+		if _, err := a.CreatePost(autoResponsePost, channel, false); err != nil {
+			l4g.Error(err.Error())
```

Comment:
You should return the error and let the caller decide what to do with it. As an example of a case when logging this isn't the right thing to do, tests that call this should assert that there is no error.

Reply:
I agree with the idea of letting the caller decide what to do with the error.

At least for this scenario, sending an autoResponder is a byproduct of a user sending a regular direct message.
I don't believe the api should return an error to the client, since this would create a side-effect in the logic for sending a notification.

That's why I used a convention MM uses when it's not necessary to handle an error `l4g.Error(err.Error())`

If this is not the preferred convention let me know, I could return the error and log the error from the caller code if that's preferred?

For reference, I've noticed many chunks of code follow the same convention I used.
* https://github.com/mattermost/mattermost-server/blob/master/app/notification.go#L97
* https://github.com/mattermost/mattermost-server/blob/master/app/notification.go#L610